flutter_todo/lib/components/app_bar.dart

```diff
@@ -1,34 +1,43 @@
 // ignore_for_file: prefer_const_constructors
 
 import 'package:flutter/material.dart';
-import 'package:provider/provider.dart';
-import 'package:realm/realm.dart';
 import 'package:flutter_todo/realm/app_services.dart';
+import 'package:flutter_todo/realm/realm_services.dart';
+import 'package:provider/provider.dart';
+import 'package:flutter_todo/realm/state_services.dart';
 
 class TodoAppBar extends StatelessWidget with PreferredSizeWidget {
-  const TodoAppBar({Key? key}) : super(key: key);
+  TodoAppBar({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     final app = Provider.of<AppServices>(context);
+    final realm = Provider.of<RealmServices?>(context);
+    final state = Provider.of<StateServices>(context);
 
     Future<void> logOut() async {
       await app.logOutUser();
+      realm?.dispose();
```

Comment:
you think there could be a way to abstract these 2 method calls into a single method?

i think it would be clearer design if we could consolidate this in a single interface.

Reply:
maybe the `app.logOutUser()` could also handle call to `realm?.dispose()`?

Reply:
Done